File: lib/java/src/main/java/uapi/internal/types/UArray.java
package uapi.internal.types;

import static uapi.internal.generation.GenerateRandomArray.generateRandomArray;
import static uapi.internal.validation.ValidateArray.validateArray;

import java.util.List;

import uapi.internal.generation.GenerateContext;
import uapi.internal.validation.ValidateContext;
import uapi.internal.validation.ValidationFailure;

public class UArray implements UType {

    public static final String _ARRAY_NAME = "Array";

    @Override
    public int getTypeParameterCount() {
        return 1;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<UTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateArray(value, typeParameters, ctx);
    }

    @Override
    public Object generateRandomValue(GenerateContext ctx) {
        return generateRandomArray(ctx);
    }

    @Override
    public String getName() {
        return _ARRAY_NAME;
    }
}

File: lib/py/uapi/internal/types/UArray.py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from uapi.internal.validation.ValidateContext import ValidateContext
    from uapi.RandomGenerator import RandomGenerator
    from uapi.internal.validation.ValidationFailure import ValidationFailure
    from uapi.internal.types.UTypeDeclaration import UTypeDeclaration
    from uapi.internal.generation.GenerateContext import GenerateContext


from uapi.internal.types.UType import UType

_ARRAY_NAME = "Array"


class UArray(UType):

    def get_type_parameter_count(self) -> int:
        return 1

    def validate(self, value: object,
                 type_parameters: list['UTypeDeclaration'], ctx: 'ValidateContext') -> list['ValidationFailure']:
        from uapi.internal.validation.ValidateArray import validate_array
        return validate_array(value, type_parameters, ctx)

    def generate_random_value(self, ctx: 'GenerateContext') -> object:
        from uapi.internal.generation.GenerateRandomArray import generate_random_array
        return generate_random_array(ctx)

    def get_name(self) -> str:
        return _ARRAY_NAME
